### PR TITLE
GRID-491: Aggregations & Group View Fix

### PR DESCRIPTION
**Pull request by Dwaynekj**
@joneit 

See this [PR](https://github.com/openfin/fin-hypergrid/pull/474)
Will Need to do the usual bump package.json

### CODE REVIEW

File: js/DataSourceAggregator.js [hunk 1]
```diff
@@ -140,6 +156,7 @@ var DataSourceAggregator = Base.extend('DataSourceAggregator', {
     addAggregate: function(label, func) {
         func.header = headerify.transform(label);
         this.aggregates.push(func);
+        this.schema.push({name: func.header});
```
Comment (joneit) [inline js/DataSourceAggregator.js]: I think this should more properly be:
```javascript
this.schema.push({
    name: label,
    header: func.header
});
```

File: js/DataNodeLeaf.js [hunk 2]
```diff
@@ -51,14 +51,14 @@ var DataNodeLeaf = DataNodeBase.extend('DataNodeLeaf', {
         if (index.length) {
             var groupsOffset = Number(drillDown.hasGroups()),
```
Comment (Dwaynekj) [inline js/DataNodeLeaf.js]: Done

File: js/DataNodeLeaf.js [hunk 3]
```diff
@@ -51,14 +51,14 @@ var DataNodeLeaf = DataNodeBase.extend('DataNodeLeaf', {
...
-                dataLen = drillDown.getColumnCount() + groupsOffset,
+                dataLen = drillDown.getColumnCount() - groupsOffset,
                 i = 0,
                 sorter = drillDown.sorterInstance;
 
             sorter.index = index;
 
             for (i; i < dataLen; i++) {
-                data[groupsOffset + i] = sorter.getValue(i, 0);
+                data[i + groupsOffset] = sorter.getValue(i, 0);
```
Comment (Dwaynekj) [inline js/DataNodeLeaf.js]: Done

File: js/DataSourceAggregator.js [hunk 4]
```diff
@@ -124,12 +142,10 @@ var DataSourceAggregator = Base.extend('DataSourceAggregator', {
         if (!this.viewMakesSense()) {
             return this.dataSource.getHeaders();
         }
-        var headers = this.aggregates.map(function(e) {
-            return e.header;
+        var headers = this.schema.map(function(e) {
+            return e.name;
```
Comment (Dwaynekj) [inline js/DataSourceAggregator.js]: Done

File: js/DataSourceGroupView.js [hunk 5]
```diff
@@ -85,11 +102,10 @@ var DataSourceGroupView = Base.extend('DataSourceGroupView', {
         if (!this.viewMakesSense()) {
             return this.dataSource.getHeaders();
         }
-        var headers = this.dataSource.getHeaders().slice(0);
+        var headers = this.schema.map(function(e) {
+            return e.name;
```
Comment (Dwaynekj) [inline js/DataSourceGroupView.js]: Done

File: js/DataSourceAggregator.js (hunk 1 above)
Comment (Dwaynekj) [inline js/DataSourceAggregator.js]: Done

File: js/DataSourceAggregator.js [hunk 6]
```diff
@@ -373,7 +390,7 @@ var DataSourceAggregator = Base.extend('DataSourceAggregator', {
      */
     getGrandTotals: function() {
         var view = this.tree;
-        return [view.data];
+        return view.data;
```
Comment (Dwaynekj) [inline js/DataSourceAggregator.js]: I'll leave this to your PR

**Pull request merged**